### PR TITLE
Add tests, remove `main`, fix bug in `Sparsity`

### PR DESCRIPTION
**Pull request by Hardvan**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/test/java/com/thealgorithms/misc/SparsityTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import org.junit.jupiter.api.Test;
+
+public class SparsityTest {
+
+    private static final double DELTA = 1e-9;
+
+    @Test
+    public void testAllZeroElements() {
+        double[][] mat = {{0, 0, 0}, {0, 0, 0}, {0, 0, 0}};
```
Comment (alxkm) [inline src/test/java/com/thealgorithms/misc/SparsityTest.java:14]: Could you please refactor these tests to parametrized, to avoid code duplication

**Review by alxkm — CHANGES_REQUESTED** [review]

File: src/test/java/com/thealgorithms/misc/SparsityTest.java (hunk 1 above)
Comment (Hardvan) [reply src/test/java/com/thealgorithms/misc/SparsityTest.java]: Refactored the test cases to Parameterized Tests!

**Comment by alxkm** [comment]
@Hardvan to merge this, update issues should be resolved

**Comment by Hardvan** [comment]
@alxkm I have resolved the merge conflict.

**Review by alxkm — APPROVED** [review]

**Pull request merged**